node/blockchain/utils.py

```diff
@@ -0,0 +1,19 @@
+from contextlib import ContextDecorator
+
+from node.core.exceptions import BlockchainLockedError
+
+
+class TableLocker(ContextDecorator):
+
+    def __init__(self, model):
+        self.db_table_name = model._meta.db_table
+        self.table_lock = None
+
+    def __enter__(self):
+        from node.blockchain.models import TableLock
+        self.table_lock, created = TableLock.objects.get_or_create(name=self.db_table_name)
+        if not created:
+            raise BlockchainLockedError('Blockchain locked. Probably it is being modified by another process')
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.table_lock.delete()
```

Comment:
does this validate that lock exists? what if someone has deleted it by now

Reply:
I added code to validate that lock object existed and was deleted correctly.